src/intlFormatDistance/index.ts

```diff
@@ -0,0 +1,149 @@
+ * depending on the distance (the less the distance the smaller the unit),
+ * or allowes a user to pass in a unit as well.
+ *
+ * | Distance between the below dates, no unit passed in:         | Result        |
+ * | new Date(1986, 3, 4, 11, 30, 0),                             | in 1 hour     |
+ * | new Date(1986, 3, 4, 10, 30, 0)                              |               |
+*  |--------------------------------------------------------------|---------------|
+ * | Distance between the below dates, hour passed in as a unit:  | Result        |
+ * | new Date(1986, 3, 4, 11, 30, 0),                             | in 60 minutes |
+ * | new Date(1986, 3, 4, 10, 30, 0)                              |               |
+
+ * @param {Date|Number} date - the date
+ * @param {Date|Number} baseDate - the date to compare with.
+ * @param {Object} [rtfOptions] - an object with options.
+ * @param {String} [rtfOptions.unit=''] - values "year", "quarter", "month", "week", "day", "hour", "minute", "second" as per MDN Web Docs.
+ * @param {String|String[]} [rtfOptions.locales='en'] - a string (or array of strings) with a BCP 47 language tag. For reference see:
+ * https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl#locale_identification_and_negotiation
+ * @param {String} [rtfOptions.options.localeMatcher='best fit'] - the locale matching algorithm to use. Other value: "lookup".
+ * @param {String} [rtfOptions.options.numeric='always'] - the output message format. Other value: "auto".
+ * @param {String} [rtfOptions.options.style='long'] - the length of the internationalized message. Other values: "short" or "narrow";
+ * For reference see:
+ * https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/RelativeTimeFormat/RelativeTimeFormat
+ *
+ * @example
+ * // What is the distance between Mar, 4 1986 11:30:00 and Mar, 4 1986 10:30:00 in Intl?
+ * const result = intlFormatDistance(
+ *   new Date(1986, 3, 4, 11, 30, 0),
+ *   new Date(1986, 3, 4, 10, 30, 0));
+ * // => in 1 hour
+ *
+ * @example
+ * // What is the distance between Mar, 4 1986 11:30:00 and Mar, 4 1986 10:30:00 in minutes in Intl?
+ * const result = intlFormatDistance(
+ *   new Date(1986, 3, 4, 11, 30, 0),
+ *   new Date(1986, 3, 4, 10, 30, 0),
+ *   { unit: 'minute'});
+ * // => in 60 minutes
+ *
+ * @example
+ * // What is the distance between Mar, 4 1986 11:30:00 and Mar, 4 1986 10:30:00 in minutes in Spanish in Intl?
+ * const result = intlFormatDistance(
+ *   new Date(1986, 3, 4, 11, 30, 0),
+ *   new Date(1986, 3, 4, 10, 30, 0),
+ *   { unit: 'minute', locales: 'es' });
+ * // => dentro de 60 minutos
+ */
+
+// This type mimics the 'options' parameter of Intl.RelativeTimeFormat
+type IrtfOptions = {
+  localeMatcher?: String,
+  numeric?: String,
+  style?: String,
+};
+
+type Options = {
+  unit?: String,
+  locales?: String;
+  options?: IrtfOptions;
+}
```

Comment:
Please upgrade typescript to the latest (4.1.3) and add `ESNext.Intl` to https://github.com/date-fns/date-fns/blob/master/tsconfig.json#L4

```suggestion
interface Options {
  localeMatcher?: Intl.RelativeTimeFormatLocaleMatcher,
  numeric?: Intl.RelativeTimeFormatNumeric,
  style?: Intl.RelativeTimeFormatStyle,
  unit?: Unit,
  locale?: Intl.RelativeTimeFormatStyle
}
```

Note that I renamed `locales` -> `locale` as I think a single locale will be used the most, so better make it look good for the default case.

Then add `Unit` to https://github.com/date-fns/date-fns/blob/master/src/types.ts:

```ts
export type Unit =
  | 'year'
  | 'quarter'
  | 'month'
  | 'week'
  | 'day'
  | 'hour'
  | 'minute'
  | 'second'
```

There's `RelativeTimeFormatUnit` but I prefer limit options (it's easy to extend than limit later) and use the same type in other functions.

Reply:
@kossnocorp 
the only thing i decided to change was the order or the properties (unit goes first) to follow the order in the MDN documentation (if thats ok). I think this makes more sense :

```
interface Options {
  unit?: Unit
  locale?: Intl.RelativeTimeFormatStyle
  localeMatcher?: Intl.RelativeTimeFormatLocaleMatcher
  numeric?: Intl.RelativeTimeFormatNumeric
  style?: Intl.RelativeTimeFormatStyle
}


```